### PR TITLE
Resolve all 13 deferred structural findings from the v0.10.0 audit (#1096)

### LINKED ISSUE #1096: v0.10.0 pre-release audit: deferred structural findings
The v0.10.0 pre-release Phase-1 audit (v0.9.2..HEAD, 84 findings, 0 blockers) surfaced these **structural / behavioral** items that were intentionally deferred from the cheap-fix PRs ([#1093](https://github.com/nugget/thane-ai-agent/pull/1093) docs, [#1094](https://github.com/nugget/thane-ai-agent/pull/1094) code-hygiene, [#1095](https://github.com/nugget/thane-ai-agent/pull/1095) model-facing) because each is an API-signature change, a denormalization removal, or a design coupling that deserves its own change rather than a mechanical edit.

Each was adversarially verified during the audit. None blocks the v0.10.0 cut.

## Loop runtime
- [ ] **`IterationResult.Supervisor` / `IterationSnapshot.Supervisor` denormalize `SupervisorTrigger`** — `internal/runtime/loop/config.go:466-475`, `internal/runtime/loop/loop.go:1242-1255`. Either drop the "backwards-compatible" framing from the doc (the real reason is event-payload ergonomics) or collapse the bool and have event/metacognitive consumers test `SupervisorTrigger` directly.
- [ ] **`BuildLoopConfig` (ego, metacog, archivist) and `BuildSpec` (ego, archivist) are exported with zero production callers**, and `BuildLoopConfig`'s doc rationale ("compatibility shim for callers that work directly with loop.Config") is false. `internal/runtime/ego/ego.go:162`, `internal/runtime/metacognitive/metacognitive.go:244`, `internal/runtime/archivist/archivist.go:200`. Delete or justify. (Relates to the #1086 envelope work.)
- [ ] **`WakeTagsFromContext` has no callers anywhere** (incl. tests) — `internal/runtime/loop/signals.go:82`. Delete or wire up.
- [ ] **`WatchedSubscription` is an adapter shim over `loop.EntitySubscription`** — parallel representations bridged at render time. `internal/state/awareness/loop_subscription_provider.go:185`. Collapse if the second representation isn't earning its keep.

## Model-facing (HA)
- [ ] **`ha_search_states` / `ha_list_entities` drop the recency delta** that the canonical entity contextfmt projection always carries — `internal/tools/ha_search_states.go:201-215`. Add a delta-formatted `since`/`updated` (via `promptfmt.FormatDeltaOnly` from `state.LastChanged/LastUpdated`) so the model gets the same recency signal across surfaces.
- [ ] **`entity_metadata.DefaultContext` is a redundant parallel encoding of `ContextRole`** — `internal/integrations/homeassistant/entity_metadata.go:100`. Drop the field (and `entityDefaultContext`) and let `context_role` carry the signal; derive a bool at the read site if some surface needs one.
- [ ] **`Floor` and `Building` are mutually-exclusive same-typed pointers** that can't be documented stand-alone — `internal/integrations/homeassistant/entity_metadata.go:114-116`. Prefer one floor slot + a label/alias indicator, or (if the JSON contract must stay) add field docs stating the mutual exclusivity.

## Code-path / API shape
- [ ] **`SetWorkingMemoryStore` must be called before `SetArchiveStore`** or the working-memory search surface is silently dropped from `archive_search` with no error — `internal/tools/working_memory_tools.go:12`. Make it order-independent (lazy read of `workingMemoryStore` at search time, or a getter into `NewMemorySearch`), or assert/log when nil at compose time.
- [ ] **`NewTools` takes a variadic `*SubscriptionStore` but only uses the first**, with one caller that always supplies it — `internal/integrations/forge/tools.go:31`. Make it a plain required parameter to remove the silent-drop footgun.
- [ ] **`agent/loop.go` `history` parameter is ignored but still passed by three callers** — `internal/runtime/agent/loop.go:1018-1028`. Drop it from the three signatures + call sites + tests so the signature reflects that history no longer shapes the system prompt.
- [ ] **`CoreContextProvider.TagContext` / `TagContextBucket` are dead** after the core-rendering refactor — `internal/runtime/agent/core_context.go:141-147`. Delete (plus the TagContext-only helpers) or document why the provider path is retained.
- [ ] **`catalog_tags` `Parents` field is authored on every leaf but has no production consumer** — `internal/model/toolcatalog/catalog_tags.go:54`. Wire the hierarchical renderer to it, or soften the doc to "reserved" until it ships.

## Docs
- [ ] **`configuration.md` Document Roots section teaches the deprecated `paths:` / `doc_roots:` split and never mentions the unified `roots:` block** — `docs/operating/configuration.md:183-244`. Rewrite to teach `roots:` (the `paths:`/`doc_roots:` keys are retained-but-deprecated).

---
_Filed from the v0.10.0 pre-release Phase-1 audit. The cheap/mechanical findings shipped in #1093/#1094/#1095._

### PR DESCRIPTION
**Pull request by thane-developer**
Closes #1096.

The v0.10.0 pre-release audit deferred 13 structural/behavioral findings. An audit against current code confirmed **all 13 were still outstanding** (none had been picked up). This PR resolves every one, grouped into five commits by kind.

## Loop runtime
- [x] **LR1** — `IterationResult`/`IterationSnapshot.Supervisor`: reworded the false "backwards-compatible" doc to state the real reason (a convenience projection of `SupervisorTrigger != ""` for event/wire consumers). Kept the bool (it's on the JSON wire).
- [x] **LR2** — deleted the zero-caller `BuildLoopConfig` (ego/archivist/metacog) + `BuildSpec` (ego/archivist). Production registers via `DefinitionSpec` + `HydrateSpec` and applies `Profile` at request time, so the manual merge was a dead duplicate. Metacog tests re-pointed to the live `HydrateSpec(DefinitionSpec(...))` path.
- [x] **LR3** — deleted the zero-caller `WakeTagsFromContext` getter (the `withWakeTags` write path stays live).
- [x] **LR4** — `WatchedSubscription` isn't a migration-in-flight shim; it's the SQLite row type for the always-visible watchlist (carries `Scope`/`ExpiresAt`), deliberately distinct from `loop.EntitySubscription`. Reworded the comments to say so.

## Model-facing (HA)
- [x] **HA1** — `ha_search_states`/`ha_list_entities` now carry the recency delta (`since`/`updated`) via a shared `haRecencyDelta` helper mirroring the canonical `contextfmt` projection.
- [x] **HA2** — dropped `EntityVisibility.DefaultContext` (+ `entityDefaultContext`), a redundant parallel encoding of `ContextRole`. Tests derive the bool from `context_role`.
- [x] **HA3** — documented that `EntityAreaMetadata.Floor`/`Building` are mutually exclusive, gated by `homeassistant.floor_alias`.

## Code-path / API shape
- [x] **CP1** — made the working-memory search surface order-independent: `archive_search` re-composes from whichever of `SetArchiveStore`/`SetWorkingMemoryStore` runs second, instead of silently freezing a nil working store.
- [x] **CP2** — `forge.NewTools` takes a plain required `*SubscriptionStore` instead of a variadic that silently used only the first element.
- [x] **CP3** — dropped the ignored `history []memory.Message` param from the three `buildSystemPrompt*` builders + all call sites (history flows as role-native messages via `buildInitialLLMMessages`). Call sites rewritten with `gofmt -r`.
- [x] **CP4** — deleted the dead `CoreContextProvider.TagContext`/`TagContextBucket` (the provider is never registered; `promptSections` is called directly).
- [x] **CP5** — marked `toolcatalog` `Parents` as reserved: authored + invariant-checked, but no renderer consumes it yet (navigation flows through `NextTags`).

## Docs
- [x] **DOC1** — rewrote the Document Roots sections of `configuration.md` and `document-roots.md` to teach the unified `roots:` block (path + policy in one entry), with `paths:`/`doc_roots:` demoted to a deprecation note.

## Test plan
- [x] `just ci` green (exit 0, 0 architecture errors).
- [x] Dead-code deletions verified zero-caller before removal; coverage preserved by re-pointing tests to live paths.
- [x] `gofmt -r` call-site rewrites verified: agent package builds + tests pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/tools/ha_entity_metadata.go [hunk 1]
```diff
@@ -30,9 +32,27 @@ type haListEntityItem struct {
...
+// the same "how fresh is this?" signal the always-on context does. since is the
+// time since the last state change; updated is set only when the last attribute
+// update meaningfully post-dates that change.
+func haRecencyDelta(s homeassistant.State, now time.Time) (since, updated string) {
+	if !s.LastChanged.IsZero() {
+		since = promptfmt.FormatDeltaOnly(s.LastChanged, now)
+	}
+	if !s.LastUpdated.IsZero() && s.LastUpdated.Sub(s.LastChanged) > time.Second {
+		updated = promptfmt.FormatDeltaOnly(s.LastUpdated, now)
+	}
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/tools/ha_entity_metadata.go]: Good catch — fixed in 8f0256d2. `haRecencyDelta` now returns neither field when `LastChanged` is zero, so `updated` can no longer populate off a nonsense `LastUpdated.Sub(zero)` delta while `since` is omitted. Added a table test covering both-set/equal/differing plus the zero-`LastChanged` edge case.

**Review by nugget — APPROVED** [review]
